Error loading modules

Hi,

I'm trying to use this action but I'm getting an error loading a module. My React Typescript application is not located in the root. So I use the following argument to point in the right path:

```
 files: src/Unosquare.AssetCore.Server/client-app/src/**/*
```

It seems like eslint is finding my config file but not the related modules declared:

```
##[error]Failed to load plugin 'react' declared in 'src/Unosquare.AssetCore.Server/client-app/.eslintrc.js': Cannot find module 'eslint-plugin-react'
```

The node_modules folder is restored before to use this GH Action.

Thanks!

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -1,12 +1,14 @@
 {
   "git.ignoreLimitWarning": true,
   "eslint.enable": true,
-  "eslint.autoFixOnSave": true,
   "eslint.validate": [
     "javascript",
     "javascriptreact",
-    { "language": "typescript", "autoFix": true },
-    { "language": "typescriptreact", "autoFix": true }
+    "typescript",
+    "typescriptreact"
   ],
   "typescript.tsdk": "node_modules/typescript/lib",
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  },
 }

--- a/action.yml
+++ b/action.yml
@@ -13,6 +13,8 @@ inputs:
   ignore:
     description: "Glob pattern to ignore from linting."
     default: "**/node_modules/**"
+  working-directory:
+    description: "Set if project root is not git root"
 runs:
   using: "node12"
   main: "lib/eslint-action.js"

--- a/action.yml
+++ b/action.yml
@@ -14,7 +14,7 @@ inputs:
     description: "Glob pattern to ignore from linting."
     default: "**/node_modules/**"
   working-directory:
-    description: "Set if project root is not git root"
+    description: "Directory where eslint should execute."
 runs:
   using: "node12"
   main: "lib/eslint-action.js"

--- a/lib/eslint-action.js
+++ b/lib/eslint-action.js
@@ -51,9 +51,11 @@ const processArrayInput = (key, required = false) => {
 function lint(files) {
     const extensions = processArrayInput('extensions', true);
     const ignoreGlob = processArrayInput('ignore');
+    const cwd = core.getInput('working-directory');
     const linter = new eslint_1.default.CLIEngine({
         extensions,
         ignorePattern: ignoreGlob,
+        cwd,
     });
     return linter.executeOnFiles(files);
 }

--- a/lib/eslint-action.js
+++ b/lib/eslint-action.js
@@ -8,6 +8,9 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
         step((generator = generator.apply(thisArg, _arguments || [])).next());
     });
 };
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
 var __importStar = (this && this.__importStar) || function (mod) {
     if (mod && mod.__esModule) return mod;
     var result = {};
@@ -15,11 +18,9 @@ var __importStar = (this && this.__importStar) || function (mod) {
     result["default"] = mod;
     return result;
 };
-var __importDefault = (this && this.__importDefault) || function (mod) {
-    return (mod && mod.__esModule) ? mod : { "default": mod };
-};
 Object.defineProperty(exports, "__esModule", { value: true });
 /* eslint-disable @typescript-eslint/camelcase */
+const path_1 = __importDefault(require("path"));
 const core = __importStar(require("@actions/core"));
 const github = __importStar(require("@actions/github"));
 const eslint_1 = __importDefault(require("eslint"));
@@ -51,7 +52,14 @@ const processArrayInput = (key, required = false) => {
 function lint(files) {
     const extensions = processArrayInput('extensions', true);
     const ignoreGlob = processArrayInput('ignore');
-    const cwd = core.getInput('working-directory');
+    let cwd = core.getInput('working-directory');
+    if (cwd && !path_1.default.isAbsolute(cwd)) {
+        cwd = path_1.default.resolve(cwd);
+    }
+    else if (!cwd) {
+        cwd = process.cwd();
+    }
+    core.debug(`Starting lint engine with cwd: ${cwd}`);
     const linter = new eslint_1.default.CLIEngine({
         extensions,
         ignorePattern: ignoreGlob,

--- a/lib/fileUtils.js
+++ b/lib/fileUtils.js
@@ -29,7 +29,7 @@ exports.filterFiles = (files, globs) => {
     const filtered = micromatch_1.default(files, globs);
     for (const file of filtered) {
         if (fs_1.default.existsSync(path_1.default.resolve(file))) {
-            result.push(file);
+            result.push(path_1.default.resolve(file));
         }
     }
     return result;

--- a/src/eslint-action.ts
+++ b/src/eslint-action.ts
@@ -41,10 +41,12 @@ const processArrayInput = (key: string, required = false): string[] => {
 function lint(files: string[]): CLIEngine.LintReport {
   const extensions = processArrayInput('extensions', true);
   const ignoreGlob = processArrayInput('ignore');
+  const cwd = core.getInput('working-directory');
 
   const linter = new eslint.CLIEngine({
     extensions,
     ignorePattern: ignoreGlob,
+    cwd,
   });
 
   return linter.executeOnFiles(files);

--- a/src/eslint-action.ts
+++ b/src/eslint-action.ts
@@ -1,4 +1,6 @@
 /* eslint-disable @typescript-eslint/camelcase */
+import path from 'path';
+
 import * as core from '@actions/core';
 import * as github from '@actions/github';
 import eslint, { CLIEngine } from 'eslint';
@@ -41,7 +43,15 @@ const processArrayInput = (key: string, required = false): string[] => {
 function lint(files: string[]): CLIEngine.LintReport {
   const extensions = processArrayInput('extensions', true);
   const ignoreGlob = processArrayInput('ignore');
-  const cwd = core.getInput('working-directory');
+  let cwd = core.getInput('working-directory');
+
+  if (cwd && !path.isAbsolute(cwd)) {
+    cwd = path.resolve(cwd);
+  } else if (!cwd) {
+    cwd = process.cwd();
+  }
+
+  core.debug(`Starting lint engine with cwd: ${cwd}`);
 
   const linter = new eslint.CLIEngine({
     extensions,

--- a/src/fileUtils.ts
+++ b/src/fileUtils.ts
@@ -13,7 +13,7 @@ export const filterFiles = (files: string[], globs: string[]): string[] => {
 
   for (const file of filtered) {
     if (fs.existsSync(path.resolve(file))) {
-      result.push(file);
+      result.push(path.resolve(file));
     }
   }
 

